code/ship/ship.cpp

```diff
@@ -660,373 +658,1053 @@ int lightningtype_match(char *p)
+
+	memcpy(&thruster_flame_info, &other.thruster_flame_info, sizeof(thrust_pair));
+	memcpy(&thruster_glow_info, &other.thruster_glow_info, sizeof(thrust_pair));
+	memcpy(&thruster_secondary_glow_info, &other.thruster_secondary_glow_info, sizeof(thrust_pair_bitmap));
+	memcpy(&thruster_tertiary_glow_info, &other.thruster_tertiary_glow_info, sizeof(thrust_pair_bitmap));
+	memcpy(&thruster_distortion_info, &other.thruster_distortion_info, sizeof(thrust_pair_bitmap));
+
+	thruster01_glow_rad_factor = other.thruster01_glow_rad_factor;
+	thruster02_glow_rad_factor = other.thruster02_glow_rad_factor;
+	thruster03_glow_rad_factor = other.thruster03_glow_rad_factor;
+	thruster02_glow_len_factor = other.thruster02_glow_len_factor;
+	thruster_dist_rad_factor = other.thruster_dist_rad_factor;
+	thruster_dist_len_factor = other.thruster_dist_len_factor;
+	thruster_glow_noise_mult = other.thruster_glow_noise_mult;
+
+	draw_distortion = other.draw_distortion;
+
+	splodeing_texture = other.splodeing_texture;
+	strcpy_s(splodeing_texture_name, other.splodeing_texture_name);
+
+	replacement_textures = other.replacement_textures;
+
+	armor_type_idx = other.armor_type_idx;
+	shield_armor_type_idx = other.shield_armor_type_idx;
+	
+	can_glide = other.can_glide;
+	glide_cap = other.glide_cap;
+	glide_dynamic_cap = other.glide_dynamic_cap;
+	glide_accel_mult = other.glide_accel_mult;
+	use_newtonian_damp = other.use_newtonian_damp;
+	newtonian_damp_override = other.newtonian_damp_override;
+
+	autoaim_fov = other.autoaim_fov;
+
+	topdown_offset_def = other.topdown_offset_def;
+	topdown_offset = other.topdown_offset;
+
+	engine_snd = other.engine_snd;
+	min_engine_vol = other.min_engine_vol;
+	glide_start_snd = other.glide_start_snd;
+	glide_end_snd = other.glide_end_snd;
+
+	ship_sounds = other.ship_sounds;
+
+	num_maneuvering = other.num_maneuvering;
+	memcpy(maneuvering, other.maneuvering, sizeof(man_thruster) * MAX_MAN_THRUSTERS);
+
+	radar_image_2d_idx = other.radar_image_2d_idx;
+	radar_color_image_2d_idx = other.radar_color_image_2d_idx;
+	radar_image_size = other.radar_image_size;
+	radar_projection_size_mult = other.radar_projection_size_mult;
+
+	memcpy(ship_iff_info, other.ship_iff_info, sizeof(int) * MAX_IFFS * MAX_IFFS);
+
+	aiming_flags = other.aiming_flags;
+	minimum_convergence_distance = other.minimum_convergence_distance;
+	convergence_distance = other.convergence_distance;
+	convergence_offset = other.convergence_offset;
+
+	emp_resistance_mod = other.emp_resistance_mod;
+
+	piercing_damage_draw_limit = other.piercing_damage_draw_limit;
+
+	damage_lightning_type = other.damage_lightning_type;
+
+	shield_impact_explosion_anim = other.shield_impact_explosion_anim;
+	hud_gauges = other.hud_gauges;
+	hud_enabled = other.hud_enabled;
+	hud_retail = other.hud_retail;
+
+	displays = other.displays;
+
+	pathMetadata = other.pathMetadata;
+
+	glowpoint_bank_override_map = other.glowpoint_bank_override_map;
+}
+
+void ship_info::move(ship_info& other)
```

Comment:
This should probably use `ship_info&&` to make sure move semantics are properly used in the function.